### PR TITLE
docs: translate to zh-CN

### PR DESCRIPTION
**Pull request by edison1105**
translate docs to Chinese.
Welcome to calibration and review.

### CODE REVIEW

**Comment by posva** [comment]
Oh, nice! There is a lot of traffic from China in the docs, so I think this would be nice to be integrated in the repository but it will have to be reviewed by some of the core team so it might take some time

**Comment by edison1105** [comment]
> Oh, nice! There is a lot of traffic from China in the docs, so I think this would be nice to be integrated in the repository but it will have to be reviewed by some of the core team so it might take some time


Got it.

**Comment by edison1105** [comment]
Hi @posva
This PR is readly for review.

**Comment by posva** [comment]
Thanks a lot! I will relay the message to get some reviews!

File: docs/zh/guide/advanced/composition-api.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    })
+  },
+}
+```
+
+组合式 API 守卫也可以用在任何由 `<router-view>` 渲染的组件中，它们不必像组件内守卫那样直接用在路由组件上。
+
+## `useLink`
+
+Vue Router 将 RouterLink 的内部行为作为一个组合式 API 函数公开。它提供了与 [`v-slot`API](/zh/api/#router-link-s-v-slot) 相同的访问属性：
```
Comment (antfu) [inline docs/zh/guide/advanced/composition-api.md:87]:
```suggestion
Vue Router 将 RouterLink 的内部行为作为一个组合式 API 函数公开。它提供了与 [`v-slot` API](/zh/api/#router-link-s-v-slot) 相同的访问属性：
```

File: docs/zh/guide/advanced/data-fetching.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        }
+      })
+    },
+  },
+}
+```
+
+## 在导航完成前获取数据
+
+通过这种方式，我们在导航转入新的路由前获取数据。我们可以在接下来的组件的 beforeRouteEnter 守卫中获取数据，当数据获取成功后只调用 next 方法：
```
Comment (antfu) [inline docs/zh/guide/advanced/data-fetching.md:74]:
```suggestion
通过这种方式，我们在导航转入新的路由前获取数据。我们可以在接下来的组件的 `beforeRouteEnter` 守卫中获取数据，当数据获取成功后只调用 `next` 方法：
```

File: docs/zh/guide/advanced/dynamic-routing.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+页面仍然会显示 `Article` 组件，我们需要手动调用 `router.replace()` 来改变当前的位置，并覆盖我们原来的位置（而不是添加一个新的路由，最后在我们的历史中两次出现在同一个位置）：
+
+```js
+router.addRoute({ path: '/about', component: About })
+// 我们也可以是用 this.$route or route = useRoute() （在 setup 中）
+router.replace(router.currentRoute.value.fullPath)
+```
+
+记住，如果你需要等待新的路由显示，可以 `await router.replace()`。
```
Comment (antfu) [inline docs/zh/guide/advanced/dynamic-routing.md:32]:
```suggestion
记住，如果你需要等待新的路由显示，可以使用 `await router.replace()`。
```

File: docs/zh/guide/advanced/dynamic-routing.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+
+## 在导航守卫中添加路由
+
+如果你决定在导航守卫内部添加或删除路由，你不应该调用 `router.replace()` ，而是通过返回新的位置来触发重定向：
```
Comment (antfu) [inline docs/zh/guide/advanced/dynamic-routing.md:37]:
```suggestion
如果你决定在导航守卫内部添加或删除路由，你不应该调用 `router.replace()`，而是通过返回新的位置来触发重定向：
```

File: docs/zh/guide/advanced/navigation-failures.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+# 等待导航结果
+
+当使用 `router-link` 组件时，Vue Router 会自动调用 `router.push` 来触发一次导航。虽然大多数链接的预期行为是将用户导航到一个新页面，但也有少数情况下用户将留在同一页面上：
+
+- 用户已经位于他们正在尝试导航到的页面
+- 一个 [导航守卫](./navigation-guards.md)通过调用 `return false` 中断了这次导航
+- 当前的导航守卫还没有完成时，一个新的导航守卫会出现了
+- 一个 [导航守卫](./navigation-guards.md)通过返回一个新的位置，重定向到其他地方 (例如， `return '/login'`)
```
Comment (antfu) [inline docs/zh/guide/advanced/navigation-failures.md:8]:
```suggestion
- 一个[导航守卫](./navigation-guards.md)通过调用 `return false` 中断了这次导航
- 当前的导航守卫还没有完成时，一个新的导航守卫会出现了
- 一个[导航守卫](./navigation-guards.md)通过返回一个新的位置，重定向到其他地方 (例如， `return '/login'`)
```

File: docs/zh/guide/advanced/navigation-guards.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
+# 导航守卫
+
+正如其名， vue-router 提供的导航守卫主要用来通过跳转或取消的方式守卫导航。这里有很多方式植入路由导航中：全局的，单个路由独享的， 或者组件级的。
```
Comment (antfu) [inline docs/zh/guide/advanced/navigation-guards.md:3]:
```suggestion
正如其名，vue-router 提供的导航守卫主要用来通过跳转或取消的方式守卫导航。这里有很多方式植入路由导航中：全局的，单个路由独享的，或者组件级的。
```

File: docs/zh/guide/advanced/scroll-behavior.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ left: 0, top: 0 })
+      }, 500)
+    })
+  },
+})
+```
+
+It's possible to hook this up with events from a page-level transition component to make the scroll behavior play nicely with your page transitions, but due to the possible variance and complexity in use cases, we simply provide this primitive to enable specific userland implementations.
```
Comment (antfu) [inline docs/zh/guide/advanced/scroll-behavior.md:109]: 😉

**Review by antfu — APPROVED** [review]
@posva Did a quick and rough overview on GitHub, I think the translations are quite natural and there isn't much I can complain about. Just a few nitpicking on formatting, but I think it's a pretty good starting point for the Chinese doc and ready to merge as long as the build process is alright.

@edison1105 Well done! I have left a few comments about the formatting glitches, please have a look. I think it's a bit redundant to have spaces around `。`，` since they are already full-width characters. I address a few but you might like to have a global search over that.

**Comment by edison1105** [comment]
@antfu thanks for your review.I will check the things you mentioned above in each file.

File: docs/zh/guide/advanced/dynamic-routing.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    router.addRoute(generateRoute(to))
+    // 触发重定向
+    return to.fullPath
+  }
+})
+```
+
+上面的例子有两个假设：第一，新添加的路由记录将与 `to` 位置相匹配，实际上导致与我们试图访问的位置不同。第二，`hasNecessaryRoute()` 在添加新的路由后返回 `false`，以避免无限重定向。
+
+因为我们是在重定向，所以我们是在替换正在进行的导航，实际上行为就像之前的例子一样。在实际场景中，添加更有可能发生在导航守卫之外，例如，当一个视图组件挂载时，它会注册新的路由。
```
Comment (meteorlxy) [inline docs/zh/guide/advanced/dynamic-routing.md:51]:
```suggestion
因为我们是在重定向，所以我们是在替换正在进行的导航，实际上行为就像之前的例子一样。而在实际场景中，添加路由的行为更有可能发生在导航守卫之外，例如，当一个视图组件挂载时，它会注册新的路由。
```

Comment (meteorlxy) [reply docs/zh/guide/advanced/dynamic-routing.md:51]: > 因为我们是在重定向，所以我们是在替换正在进行的导航，实际上行为就像之前的例子一样

这句感觉可以改改

File: docs/zh/introduction.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# 介绍
+
+Vue Router 是【Vue.js】(http://v3.vuejs.org)的官方路由。它与 Vue.js 核心深度集成，让用 Vue.js 构建单页应用变得轻而易举。功能包括：
```
Comment (meteorlxy) [inline docs/zh/introduction.md:3]:
```suggestion
Vue Router 是 [Vue.js](http://v3.vuejs.org)的官方路由。它与 Vue.js 核心深度集成，让用 Vue.js 构建单页应用变得轻而易举。功能包括：
```

**Review by meteorlxy — CHANGES_REQUESTED** [review]
Did a partial review

**Comment by posva** [comment]
Let's merge this to make it available for all Chinese users and they will be able to propose changes as they see.

I have seen a few usability issues when it comes to language selection but I will have to fix those in Vitepress

Thanks a lot @edison1105 @antfu @meteorlxy

**Pull request merged**